e2e/app/FeaturedWorkspacesPage.ts

```diff
@@ -0,0 +1,34 @@
+import {Page} from 'puppeteer';
+import {waitUntilFindTexts} from '../driver/waitFuncs';
+import AuthenticatedPage from './page-mixin/AuthenticatedPage';
+import {SideNav} from './page-mixin/SideNav';
+
+export const FIELD_LABEL = {
+  TITLE: 'Workspace Library',
+  HEADER: 'Researcher Workbench Workspace Library',
+};
+
+export default class FeaturedWorkspaces extends AuthenticatedPage {
+
+  public page: Page;
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+  }
+
+  public async isLoaded(): Promise<boolean> {
+    await super.isLoaded(FIELD_LABEL.TITLE);
+    await waitUntilFindTexts(this.puppeteerPage, FIELD_LABEL.HEADER);
+    return true;
+  }
+
+  public async waitForReady(): Promise<FeaturedWorkspaces> {
```

Comment:
[Potential improvement / follow-up] I see many nearly-identical implementations of this waitForReady method. This should probably be a method implemented on the base class.

Maybe the reason you didn't do it that way was because of the desire to have a specific type of component in the returned promise?

I think you can get around this in Typescript with this (pun intended): https://www.typescriptlang.org/docs/handbook/advanced-types.html#polymorphic-this-types
```
export class BasePage {
  public async waitForReady(): Promise<this> {
    await this.isLoaded();
    await this.waitForSpinner();
    return this;
  }
}
```

(Disclaimer: I haven't actually tested the above.)

Reply:
👍 Done per suggestion

Reply:
this is in `authenticated-page.ts` now.
```
abstract async isLoaded(): Promise<boolean>

async waitForReady(): Promise<this> {
    await this.isLoaded();
    await this.waitUntilNoSpinner();
    return this;
  }
```
All extending classes implement its own version of `isLoaded()`.